Kiosk/Sale Artwork Details/SaleArtworkDetailsViewController.swift

```diff
@@ -38,6 +39,25 @@ class SaleArtworkDetailsViewController: UIViewController {
         setupAdditionalDetailStackView()
     }
 
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+
+        // OK, so this is pretty weird, eh? So basically we need to be notified of layout changes _just after_ the layout
+        // is actually done. For whatever reason, the UIKit hack to get the labels to adhere to their proper width only
+        // works if we defer recalculating their geometry to the next runloop.
+        // This wasn't an issue with RAC's rac_signalForSelector because that invoked the signal _after_ this method completed.
+        // So that's what I've done here.
+        dispatch_async(dispatch_get_main_queue()) {
+            self.layoutSubviews.onNext()
+        }
+    }
+
+    override func viewWillAppear(animated: Bool) {
+        super.viewWillAppear(animated)
+
+        viewWillAppear.onCompleted()
+    }
```

Comment:
Yeah, I actually went through and this line indicates intent more clearly. The other jiggery-pokery (`self.view.setNeedsLayout` etc) actually _does_ need to be there to work :shrug: